run.c

```diff
@@ -82,86 +138,50 @@ void run(void)
 
         rebuild_activelist();
 
-        every_timestep_stuff(NumForces);	/* write some info to log-files */
+        every_timestep_stuff(NumForces, NumCurrentTiStep);	/* write some info to log-files */
 
         /* force */
-        compute_accelerations();	/* compute accelerations for
+        compute_accelerations(is_PM);	/* compute accelerations for
                                      * the particles that are to be advanced
                                      */
 
-        /* kick */
-        advance_and_find_timesteps();	/* 'kick' active particles in
-                                         * momentum space and compute new
-                                         * timesteps for them
-                                         */
+        /* 'kick' active particles in
+         * momentum space and compute new
+         * timesteps for them. If we are about to
+         * do output, only do a half kick.
+         */
+        advance_and_find_timesteps(WillOutput);
```

Comment:
Force calculation is always more expensive and we are only kicking the active particles, so it shouldn't matter?